pkg/controller/node/node_controller.go

```diff
@@ -987,28 +990,39 @@ func getCandidateMachines(pool *mcfgv1.MachineConfigPool, nodesInPool []*corev1.
 	return nodes[:capacity]
 }
 
-// getCurrentEtcdLeader is not yet implemented
-func (ctrl *Controller) getCurrentEtcdLeader(candidates []*corev1.Node) (*corev1.Node, error) {
-	return nil, nil
+// getOperatorPodNodeName fetches the name of the current node running the machine-config-operator pod
+func (ctrl *Controller) getOperatorNodeName() (string, error) {
+	// Create a selector object with  a filter on the machine-config-operator pod
+	parsedSelector, err := labels.Parse("k8s-app=machine-config-operator")
+	if err != nil {
+		glog.Infof("Fetching machine-config-operator pod selector object failed: %v", err)
+		return "", err
+	}
+	// Query pod list with selector, this slice should only have 1 element
+	podList, err := ctrl.podLister.List(parsedSelector)
+	if err != nil || len(podList) != 1 {
+		glog.Infof("Fetching machine-config-operator pod object failed: %v", err)
+		return "", err
+	}
+	return podList[0].Spec.NodeName, err
 }
 
 // filterControlPlaneCandidateNodes adjusts the candidates and capacity specifically
-// for the control plane, e.g. based on which node is the etcd leader at the time.
+// for the control plane, e.g. based on which node is running the operator node at the time.
 // nolint:unparam
 func (ctrl *Controller) filterControlPlaneCandidateNodes(pool *mcfgv1.MachineConfigPool, candidates []*corev1.Node, capacity uint) ([]*corev1.Node, uint, error) {
 	if len(candidates) <= 1 {
 		return candidates, capacity, nil
 	}
-	etcdLeader, err := ctrl.getCurrentEtcdLeader(candidates)
+	operatorNodeName, err := ctrl.getOperatorNodeName()
 	if err != nil {
 		glog.Warningf("Failed to find current etcd leader (continuing anyways): %v", err)
 	}
 	var newCandidates []*corev1.Node
 	for _, node := range candidates {
-		if node == etcdLeader {
-			// For now make this an event so we know it's working, even though it's more of a non-event
-			ctrl.eventRecorder.Eventf(pool, corev1.EventTypeNormal, "DeferringEtcdLeaderUpdate", "Deferring update of etcd leader %s", node.Name)
-			glog.Infof("Deferring update of etcd leader: %s", node.Name)
+		if node.Name == operatorNodeName {
+			ctrl.eventRecorder.Eventf(pool, corev1.EventTypeNormal, "DeferringOperatorNodeUpdate", "Deferring update of machine config operator node %s", node.Name)
+			glog.Infof("Deferring update of machine config operator node: %s", node.Name)
```

Comment:
let's move to using klog as this is what we will be using from now on in MCO for logging, see https://github.com/openshift/machine-config-operator/pull/3734